ouroboros-consensus/src/Ouroboros/Storage/ImmutableDB/Types.hs

```diff
@@ -26,44 +23,21 @@ module Ouroboros.Storage.ImmutableDB.Types
 import           Codec.Serialise (DeserialiseFailure)
 import           Control.Exception (Exception (..))
 
-import           Data.Word (Word64)
-
 import           GHC.Generics (Generic)
 import           GHC.Stack (CallStack, prettyCallStack)
 
 import           Ouroboros.Network.Block (SlotNo (..))
 
+import           Ouroboros.Storage.Common
 import           Ouroboros.Storage.FS.API.Types (FsError, FsPath, prettyFsError,
                      sameFsError)
 
-
--- | An epoch, i.e. the number of the epoch.
-newtype EpochNo = EpochNo { unEpochNo :: Word64 }
-  deriving (Eq, Ord, Enum, Num, Show, Generic)
-
-newtype EpochSize = EpochSize { unEpochSize :: Word64 }
-  deriving (Eq, Ord, Enum, Num, Show, Generic, Real, Integral)
-
--- | The offset of a slot in an index file.
-type SlotOffset = Word64
-
--- | The tip of the 'Ouroboros.Storage.ImmutableDB.API.ImmutableDB'.
--- TODO unify with Edsko's Tip:
---
--- data Tip r = Tip r | TipGen
-data Tip
```

Comment:
You could have defined three pattern synonyms for these constructors to minimise the changes :smiley: